pandas/core/frame.py

```diff
@@ -7420,23 +7421,35 @@ def _cmp_method(self, other, op):
         new_data = self._dispatch_frame_op(other, op, axis=axis)
         return self._construct_result(new_data)
 
-    def _arith_method(self, other, op):
+    def _arith_method(self, other, op, inplace: bool = False):
         if self._should_reindex_frame_op(other, op, 1, None, None):
             return self._arith_method_with_reindex(other, op)
 
         axis: Literal[1] = 1  # only relevant for Series other case
         other = ops.maybe_prepare_scalar_for_op(other, (self.shape[axis],))
 
-        self, other = self._align_for_op(other, axis, flex=True, level=None)
+        # Disable copy on write, since align will make a shallow copy
+        # (preventing it from going inplace when it should be able to)
+        # TODO: better way of doing this?
```

Comment:
cc @phofl?

Reply:
Yeah we definitely shouldn't do this. This shouldn't matter anyway? You can't do it inlace if the index of left changed and in all other cases you can simply discard left, freeing the reference